.github/workflows/i18n-weekly-release.yml

```diff
@@ -0,0 +1,55 @@
+name: I18N Weekly Release
+on:
+  schedule:
+    - cron: '0 20 * * 0'
+
+jobs:
+  i18n-release:
+    name: Release
+    runs-on: ubuntu-20.04
+
+    steps:
+      # clone the repository
+      - uses: actions/checkout@v2
+      # enable dependencies caching (vendor and node_modules are wiped during build so they are ignored here)
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/composer/
+          key: ${{ runner.os }}-composer-${{ hashFiles('composer.lock') }}
+      - uses: actions/cache@v2
+        with:
+          path: ~/.npm/
+          key: ${{ runner.os }}-npm-${{ hashFiles('package-lock.json') }}
+      # setup PHP, but without debug extensions for reasonable performance
+      - uses: shivammathur/setup-php@v2
+        with:
+          php-version: '7.4'
+          tools:       composer
+          coverage:    none
+
+      - name: Build release
+        run: |
+          npm ci
+          npm run build
+
+      - name: Upload release
+        run: |
+          TAG="i18n-$(date +%y%m%d)"
```

Comment:
I have a question related to this; are we worried about duplicate releases? Like @asumaran noticed if we try to create a release twice with the same tag we're going to get a conflict.

This might cause issues if we the action tries to create an i18n release more than once per day.

That said, this will probably not be an issue if it's only run once per week 🤔

Reply:
We're now using the full year as of f3531138c4855eeb17a9282183bce70bf621bf7f

Reply:
> This might cause issues if we the action tries to create an i18n release more than once per day.

What will happen if it runs with the same name more than once? Will the second release just fail silently?

Reply:
> are we worried about duplicate releases?

Maybe. I can only think of couple possible scenario how that happens:
1. Someone / something trigger the GH action manually before the cron.
2. Someone create the release with exact tag outside this workflow.
3. Something on GH's end runs this workflow more than once.

In any of the case, I would say the outcome isn't fatal because even though the workflow fails to complete, the release is there (and hopefully the build package is too).

Handling the error gracefully would be good though.

Also, related to this, there is this issue https://github.com/Automattic/woocommerce-payments/issues/1814 to alert us.

> What will happen if it runs with the same name more than once? Will the second release just fail silently?

At current state, the workflow will continue on, uploading build package will fail because it depends on the response of the GH API call to create the release, but it will trigger GlotPress to update translation for that release.

Reply:
> Handling the error gracefully would be good though.

We'll probably see a failed task in GitHub. Is that graceful enough?

Or should we try once more with some addition to the tag in case of a failure?

> there is issue #1814 to alert us

Maybe that's where this should be solved?

<hr />

My suggestion would be to stop the script with some sort of `exit 1` if any of the http requests fail so we can see a clear failure in the task overview.

Reply:
That's a good point. This GH action should fail in any other case than the happy case, instead of being graceful and make it look like a success. Using `exit 1` is a very good idea!

Reply:
It was a bit more involved to get the exit statements in there [given how curl handles errors](https://superuser.com/a/1641410), but I've made that change in e60c3582367094bfaa7f53ec7dcca16ff8080a64